src/sciline/pipeline.py

```diff
@@ -191,20 +190,22 @@ def build(self, tp: Type[T], /) -> Graph:
         tp:
             Type to build the graph for.
         """
-        provider: Callable[..., T]
-        provider, bound = self._get_provider(tp)
-        tps = get_type_hints(provider)
-        args = {
-            name: _bind_free_typevars(t, bound=bound)
-            for name, t in tps.items()
-            if name != 'return'
-        }
-        graph: Graph = {tp: (provider, args)}
-        try:
+        graph: Graph = {}
+        stack: List[Tuple[Type[T], Dict[TypeVar, Key]]] = [(tp, {})]
+        while stack:
+            tp, bound = stack.pop()
+            provider: Callable[..., T]
+            provider, bound = self._get_provider(tp)
```

Comment:
If `bound` is going to be retrieved by `self._get_provider`
`stack` doesn't need to remember `bound`, right...?

Reply:
Good point, fixed!